includes/translation/translation-repository.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Wporg\TranslationEvents\Translation;
+
+use DateTimeImmutable;
+
+class Translation_Repository {
+	/**
+	 * Count the number of translations made by given users before a specified datetime.
+	 *
+	 * @param int[]             $user_ids Ids of users for which to count translations.
+	 * @param DateTimeImmutable $before   Only include translations before this datetime.
+	 *
+	 * @return array Associative array with user id as key and number of translations as value.
+	 */
+	public function count_translations_before( array $user_ids, DateTimeImmutable $before ): array {
+		global $wpdb, $gp_table_prefix;
+
+		// Prevent SQL injection.
+		foreach ( $user_ids as $user_id ) {
+			if ( ! is_int( $user_id ) ) {
+				return array();
+			}
+		}
+
+		$user_ids_param = implode( ',', $user_ids );
```

Comment:
Another way how to do this with parameters is:
```suggestion
		$user_ids_param = implode( ',', array_fill( 0, count( $user_ids ), '%d' );
```